packages/store-sync/src/fromEventSource.ts

```diff
@@ -0,0 +1,10 @@
+import { Observable } from "rxjs";
+
+export function fromEventSource<T>(url: string | URL): Observable<MessageEvent<T>> {
+  return new Observable<MessageEvent>((subscriber) => {
+    const eventSource = new EventSource(url);
+    eventSource.onmessage = (ev): void => subscriber.next(ev);
+    eventSource.onerror = (): void => subscriber.error(new Error("Event source closed: " + url));
```

Comment:
I am pretty sure `EventSource` attempts to reconnect during an error (I think indefinitely) but unclear if it emits `onerror` during this reconnect flow. We might want to give it a bad URL and see if this emits a bunch of `onerror` and maybe allow a few reconnection attempts before giving up and closing the stream with `subscriber.error`?

Reply:
Just tried that, it does indeed try to reconnect, but it seems like once it reconnects it skips the blocks that happened while the connection was down and just continues with the current block. I think we can solve this on the indexer side by using the block number as event id and also add code to handle reconnections with `last_event_id`, but would leave that as an optimization for later.

Reply:
yeah exactly, we need to update the indexer SSE stream to include an `id` and use the `Last-Event-ID` header for reconnections

Reply:
block number would likely need to be block number + log index